### ISSUE TITLE
gnorm

### ISSUE THREAD

**Issue by xiaoshengjun** [pytorch/fairseq#1263]
hi， when I training a model， it shows ：loss=1.112, ppl=2.16, wps=80656, ups=8.4, wpb=8857, bsz=510, num_updates=279614, lr=0.000332268, gnorm=1.152, clip=2%, oom=0, loss_scale=0.031, wall=8339, train_wall=26839，

What does the parameter ‘gnorm’ mean? Thank you very much.

**Comment by ngoyal2707** [comment]
L2 norm of the gradients
